### DISCUSSION TITLE
How can I plug in apollo/graphql/mongoose if "A custom server can not be deployed on Vercel, the platform Next.js was made for."?

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#17353] [Help] [answered]
Hi everyone! I'm new to Next.js, just following a Udemy course. 

It seems like doing most of the backend stuff (apollo/graphql/mongoose/etc) requires creating a custom server.

But [this page](https://nextjs.org/docs/advanced-features/custom-server) tells me that I can't use a custom server with Vercel.

What should I be using instead? How is this supposed to work on vercel?

**Comment by lfades** [comment, marked as answer]
Next.js works out of the box with Vercel if you're not using a custom server. So the solution here if you want to use Vercel is to not use a custom server.

> It seems like doing most of the backend stuff (apollo/graphql/mongoose/etc) requires creating a custom server.

You can still use Apollo, Mongoose and GraphQL without a custom server by using serverless functions instead. If your GraphQL server is huge and not suitable for a serverless function then you may want to consider an external service to deploy your GraphQL server and then deploy the Next.js app to Vercel without the custom server, that way you'll be able to use it as a frontend app that connects to your external GraphQL server.
